src/player.ts

```diff
@@ -419,10 +426,11 @@ export default class Player extends EventEmitter {
     }
   }
 
-  receiveBuildingTriggerIncome(building: Building, planet: Planet) {
+  receiveBuildingTriggerIncome(building: Building, planet: Planet, isAdditionalMine: boolean) {
     // this is for roundboosters, techtiles and adv tile
     for (const event of this.events[Operator.Trigger]) {
-      if (Condition.matchesBuilding(event.condition, building, planet)) {
+      // only new mine trigger event for Lantids in other's planet
+      if (Condition.matchesBuilding(event.condition, building, planet) && (!isAdditionalMine || (isAdditionalMine && event.condition === Condition.Mine))) {
```

Comment:
`&& (!isAdditionalMine || (isAdditionalMine && event.condition === Condition.Mine))` can be replaced by `&& (!isAdditionalMine || event.condition === Condition.Mine)`